tesseract_common/include/tesseract_common/serialization.h

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file serialization.h
+ * @brief Additional Boost serialization wrappers
+ * @details Supports the following
+ *            - Eigen::VectorXd
+ *            - Eigen::Isometry3d
+ *            - Eigen::MatrixX2d
+ *
+ * @author Levi Armstrong
+ * @date February 21, 2021
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_COMMON_SERIALIZATION_H
+#define TESSERACT_COMMON_SERIALIZATION_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <Eigen/Dense>
+#include <boost/serialization/base_object.hpp>
+#include <boost/serialization/nvp.hpp>
+#include <boost/serialization/array.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+namespace boost
+{
+namespace serialization
+{
+/*****************************/
+/****** Eigen::VectorXd ******/
+/*****************************/
+template <class Archive>
+inline void save(Archive& ar, const Eigen::VectorXd& g, const unsigned int /*version*/)
```

Comment:
Given that this file being correct is core to just about every package working correctly, it would be nice to have unit tests to test each of these types thoroughly.

Reply:
Updated to include unit tests for VectorXd, MatrixX2d and Isometry3d.